extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js

```diff
@@ -167,6 +167,14 @@ function tests(name) {
       }, null)).to.be.rejectedWith(/type/);
     });
 
+    it('should reject unknown type', () => {
+      return expect(getAd({
+        width: 300,
+        height: 250,
+        type: 'unknownType',
+      }, null)).to.be.rejectedWith(/unknownType/);
```

Comment:
canonical should be `undefined` otherwise this could also be rejected by none canonical.